syntax = "proto3";

package automatedcodingworkflow;

service AutomatedCodingWorkflowService {
    rpc StartWorkflow (StartWorkflowRequest) returns (StartWorkflowResponse);
    rpc GetWorkflowConfig (GetWorkflowConfigRequest) returns (GetWorkflowConfigResponse);
}

message StartWorkflowRequest {
}

message StartWorkflowResponse {
    string result = 1;
}

message GetWorkflowConfigRequest {
}

message GetWorkflowConfigResponse {
    repeated Stage stages = 1;
}

message Stage {
    string name = 1;
    string class_name = 1;
    repeated Stage subStages = 2;
}
